File: src/types/cliente.ts
export interface Cliente {
  id: string;
  codice: string | null;
  nome: string;
  cognome: string;
  sesso: 'M' | 'F' | 'A' | null;
  data_nascita: string | null;
  codice_fiscale: string | null;
  telefono: string | null;
  cellulare: string | null;
  email: string | null;
  indirizzo: string | null;
  citta: string | null;
  cap: string | null;
  provincia: string | null;
  consenso_marketing: boolean;
  consenso_whatsapp: boolean;
  consenso_email: boolean;
  canale_preferito: 'whatsapp' | 'email' | 'telefono' | null;
  tipo_pelle: string | null;
  allergie: string | null;
  patologie: string | null;
  note_estetiche: string | null;
  fonte_acquisizione: string | null;
  operatrice_riferimento_id: string | null;
  data_consenso_privacy: string | null;
  data_ultimo_aggiornamento_privacy: string | null;
  attivo: boolean;
  note: string | null;
  created_at: string;
  updated_at: string;
}

export interface CreateClienteInput {
  nome: string;
  cognome: string;
  sesso?: 'M' | 'F' | 'A';
  data_nascita?: string;
  cellulare?: string;
  email?: string;
  indirizzo?: string;
  note?: string;
  consenso_marketing: boolean;
  consenso_whatsapp: boolean;
  consenso_email: boolean;
  canale_preferito?: 'whatsapp' | 'email' | 'telefono';
}

export interface UpdateClienteInput {
  nome?: string;
  cognome?: string;
  data_nascita?: string;
  cellulare?: string;
  email?: string;
  indirizzo?: string;
  citta?: string;
  note?: string;
  consenso_marketing?: boolean;
  consenso_whatsapp?: boolean;
  consenso_email?: boolean;
}

File: src-tauri/src/models/cliente.rs
use serde::{Deserialize, Serialize};
use sqlx::FromRow;
use chrono::{DateTime, NaiveDate, Utc};

#[derive(Debug, Serialize, Deserialize, FromRow, Clone)]
pub struct Cliente {
    pub id: String,
    pub codice: Option<String>,

    // Dati anagrafici
    pub nome: String,
    pub cognome: String,
    pub sesso: Option<String>,
    pub data_nascita: Option<NaiveDate>,
    pub codice_fiscale: Option<String>,

    // Contatti
    pub telefono: Option<String>,
    pub cellulare: Option<String>,
    pub email: Option<String>,

    // Indirizzo
    pub indirizzo: Option<String>,
    pub citta: Option<String>,
    pub cap: Option<String>,
    pub provincia: Option<String>,

    // Preferenze comunicazione
    pub consenso_marketing: bool,
    pub consenso_sms: bool,
    pub consenso_whatsapp: bool,
    pub consenso_email: bool,
    pub canale_preferito: Option<String>,

    // Info estetiche
    pub tipo_pelle: Option<String>,
    pub allergie: Option<String>,
    pub patologie: Option<String>,
    pub note_estetiche: Option<String>,

    // Metadata
    pub fonte_acquisizione: Option<String>,
    pub operatrice_riferimento_id: Option<String>,

    // Privacy
    pub data_consenso_privacy: Option<DateTime<Utc>>,
    pub data_ultimo_aggiornamento_privacy: Option<DateTime<Utc>>,

    // Stato
    pub attivo: bool,
    pub note: Option<String>,

    // Timestamps
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct CreateClienteInput {
    pub nome: String,
    pub cognome: String,
    pub sesso: Option<String>,
    pub data_nascita: Option<NaiveDate>,
    pub cellulare: Option<String>,
    pub email: Option<String>,
    pub note: Option<String>,
    pub consenso_marketing: bool,
    pub consenso_sms: bool,
    pub consenso_whatsapp: bool,
    pub consenso_email: bool,
    pub canale_preferito: Option<String>,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct UpdateClienteInput {
    pub nome: Option<String>,
    pub cognome: Option<String>,
    pub data_nascita: Option<NaiveDate>,
    pub cellulare: Option<String>,
    pub email: Option<String>,
    pub indirizzo: Option<String>,
    pub citta: Option<String>,
    pub note: Option<String>,
}
